### PR TITLE
Clarify the key object definition for the key management API

### PR DESCRIPTION
**Pull request by turt2live**
Fixes https://github.com/matrix-org/matrix-doc/issues/1907

### CODE REVIEW

**Comment by turt2live** [comment]
It's very annoying that Swagger doesn't support the thing we want to do (anyOf on a response schema). It supports it just fine for request parameters, but not in the response schema for some reason.

Error is the cryptic:
```
JSON_OBJECT_VALIDATION_FAILED",
  "stack": "SyntaxError: Swagger schema validation failed.
  Data does not match any schemas from 'oneOf' at #/paths//keys/claim/post/responses/200
    Data does not match any schemas from 'oneOf' at #/paths//keys/claim/post/responses/200/schema
      Data does not match any schemas from 'anyOf' at #/schema/properties/one_time_keys/additionalProperties
        Data does not match any schemas from 'anyOf' at #/properties/one_time_keys/additionalProperties/additionalProperties
          Data does not match any schemas from 'anyOf' at #/additionalProperties/type
            No enum match for: string,[object Object] at #/type
            No enum match for: {\"type\":\"object\",\"title\":\"KeyObject\",\"properties\":{\"key\":{\"type\":\"string\",\"description\":\"The key, encoded using unpadded base64.\"},\"signatures\":{\"type\":\"object\",\"description\":\"Signature for the device. Mapped from user ID to signature object.\",\"additionalProperties\":{\"type\":\"string\"}}},\"required\":[\"key\",\"signatures\"]} at #/type/1
          Expected type boolean but found type object at #/additionalProperties
        Expected type boolean but found type object at #/properties/one_time_keys/additionalProperties
      Additional properties not allowed: properties at #/schema
    Missing required property: $ref at #/paths//keys/claim/post/responses/200
```

**Comment by ara4n** [comment]
errrm... looks like swagger. maybe? possibly? </dog_goggles>

**Comment by turt2live** [comment]
It's definitely swagger, although the diff is (and swagger, annoyingly) is very confusing. It's supposed to be fix the nesting levels while also making documentation a thing.

**Comment by uhoreg** [comment]
Something isn't happy in CS `/keys/upload`:
image

File: api/client-server/keys.yaml [hunk 1]
```diff
@@ -63,18 +63,41 @@ paths:
...
-                    key: "zKbLg+NrIjpnagy+pIY6uPL4ZwEG2v+8F9lmgsnlZzs"
-                    signatures:
-                      "@alice:example.com":
-                        ed25519:JLAFKJWSCS: "FLWxXqGbwrb8SM3Y795eB6OA8bwBcoMZFXBqnTn58AYWZSqiD45tlBVcDa2L7RwdKXebW/VzDlnfVJ+9jok1Bw"
-                  signed_curve25519:AAAAHQ:
-                    key: "j3fR3HemM16M7CWhoI4Sk5ZsdmdfQHsKL1xuSft6MSw"
-                    signatures:
-                      "@alice:example.com":
-                        ed25519:JLAFKJWSCS: "IQeCEPb9HFk217cU9kw9EOiusC6kMIkoIRnbnfOh5Oc63S1ghgyjShBGpu34blQomoalCyXWyhaaT3MrLZYQAA"
+                    # XXX: We can't define an actual object here, so we have to hope
```
Comment (uhoreg) [inline api/client-server/keys.yaml]: Can we add a link in the description field (line 59) to #key-algorithms, which describes the formats of the keys?  (same goes with the corresponding federation swagger file)

**Pull request merged**